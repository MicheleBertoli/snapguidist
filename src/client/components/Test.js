import React, { Component, PropTypes } from 'react'
import classnames from 'classnames'

class Test extends Component {

  constructor(props) {
    super(props)

    this.state = { expanded: false }

    this.handleClick = this.handleClick.bind(this)
    this.toggle = this.toggle.bind(this)
  }

  handleClick() {
    if (this.state.expanded) {
      this.toggle()
    }

    this.props.onClick()
  }

  toggle() {
    this.setState({ expanded: !this.state.expanded })
  }

  isPass() {
    return this.props.response && this.props.response.pass
  }

  isFail() {
    return this.props.response && !this.props.response.pass
  }

  render() {
    const containerClasses = classnames(
      'snapguidist__test',
      { 'snapguidist__test--pass': this.isPass() },
      { 'snapguidist__test--fail': this.isFail() },
      { 'snapguidist__test--expanded': this.state.expanded },
    )

    const arrowClasses = classnames(
      'snapguidist__arrow',
      { 'snapguidist__arrow--expanded': this.state.expanded },
    )

    return (
      <div className={containerClasses}>
        {this.isFail() &&
          <div>
            <div>
              <button
                className="snapguidist__button"
                disabled={this.props.isFetching}
                onClick={this.handleClick}
              >
                Update
              </button>
              <button
                className="snapguidist__button"
                disabled={this.props.isFetching}
                onClick={this.toggle}
              >
                <span className={arrowClasses}>â–¼</span>
              </button>
            </div>
            <div>
              <div className="ReactStyleguidist-common__font snapguidist__label">Actual</div>
              <code className="snapguidist__code">{this.props.response.actual}</code>
              <div className="ReactStyleguidist-common__font snapguidist__label">Expected</div>
              <code className="snapguidist__code">{this.props.response.expected}</code>
            </div>
          </div>
        }
      </div>
    )
  }

}

Test.propTypes = {
  isFetching: PropTypes.bool,
  onClick: PropTypes.func,
  response: PropTypes.shape({
    actual: PropTypes.string,
    count: PropTypes.number,
    expected: PropTypes.string,
    pass: PropTypes.bool,
  }),
}

export default Test
