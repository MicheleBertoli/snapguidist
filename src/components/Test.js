import React, { Component, PropTypes } from 'react'
import classnames from 'classnames'
import Code from './Code'

class Test extends Component {

  constructor(props) {
    super(props)

    this.state = { expanded: false }

    this.handleClick = this.handleClick.bind(this)
    this.toggle = this.toggle.bind(this)
  }

  handleClick() {
    if (this.state.expanded) {
      this.toggle()
    }

    this.props.onClick()
  }

  toggle() {
    this.setState({ expanded: !this.state.expanded })
  }

  isPass() {
    return this.props.response && this.props.response.pass
  }

  isFail() {
    return this.props.response && !this.props.response.pass
  }

  render() {
    const containerClasses = classnames(
      'snapguidist__test',
      { 'snapguidist__test--pass': this.isPass() },
      { 'snapguidist__test--fail': this.isFail() },
      { 'snapguidist__test--expanded': this.state.expanded }
    )

    const arrowClasses = classnames(
      'snapguidist__arrow',
      { 'snapguidist__arrow--expanded': this.state.expanded }
    )

    return (
      <div className={containerClasses}>
        {this.isFail() &&
          <div>
            <div>
              <button
                className="snapguidist__button"
                disabled={this.props.isQueuing}
                onClick={this.handleClick}
              >
                Update
              </button>
              <button
                className="snapguidist__button"
                disabled={this.props.isQueuing}
                onClick={this.toggle}
              >
                <span className={arrowClasses}>â–¼</span>
              </button>
            </div>
            <div>
              {this.props.response.diff
                ? <Code diff label="Difference" value={this.props.response.diff} />
                : (
                  <div>
                    <Code label="Actual" value={this.props.response.actual} />
                    <Code label="Expected" value={this.props.response.expected} />
                  </div>
                )
              }
            </div>
          </div>
        }
      </div>
    )
  }

}

Test.propTypes = {
  isQueuing: PropTypes.bool,
  onClick: PropTypes.func,
  response: PropTypes.shape({
    actual: PropTypes.string,
    count: PropTypes.number,
    diff: PropTypes.string,
    expected: PropTypes.string,
    pass: PropTypes.bool,
  }),
}

export default Test
