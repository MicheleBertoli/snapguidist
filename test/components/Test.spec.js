import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Test from '../../src/components/Test'

const props = {
  isFetching: false,
  onClick: () => {},
  response: {
    actual: 'actual',
    count: 0,
    expected: 'expected',
    pass: false,
  },
}

test('works when pass', () => {
  const component = renderer.create(<Test {...props} response={{ pass: true }} />)
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})

test('works when fail, with diff', () => {
  const wrapper = shallow(<Test {...props} />)

  expect(toJSON(wrapper)).toMatchSnapshot()
})

test('works when fail, without diff)', () => {
  const wrapper = shallow(<Test {...props} response={{ diff: 'diff' }} />)

  expect(toJSON(wrapper)).toMatchSnapshot()
})

test('works when is fetching', () => {
  const wrapper = shallow(<Test {...props} isFetching />)

  expect(toJSON(wrapper)).toMatchSnapshot()
})

test('works on update', () => {
  const onClick = jest.fn()
  const wrapper = shallow(<Test {...props} onClick={onClick} />)
  wrapper.find('button').at(0).simulate('click')

  expect(onClick).toHaveBeenCalled()
  expect(toJSON(wrapper)).toMatchSnapshot()
})

test('works on toggle', () => {
  const wrapper = shallow(<Test {...props} />)
  wrapper.find('button').at(1).simulate('click')

  expect(toJSON(wrapper)).toMatchSnapshot()
})

