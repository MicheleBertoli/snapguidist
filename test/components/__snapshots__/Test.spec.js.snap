// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works on toggle 1`] = `
<div
  className="snapguidist__test snapguidist__test--fail snapguidist__test--expanded"
>
  <div>
    <div>
      <button
        className="snapguidist__button"
        disabled={false}
        onClick={[Function]}
      >
        Update
      </button>
      <button
        className="snapguidist__button"
        disabled={false}
        onClick={[Function]}
      >
        <span
          className="snapguidist__arrow snapguidist__arrow--expanded"
        >
          ▼
        </span>
      </button>
    </div>
    <div>
      <div>
        <Code
          label="Actual"
          value="actual"
        />
        <Code
          label="Expected"
          value="expected"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`works on update 1`] = `
<div
  className="snapguidist__test snapguidist__test--fail"
>
  <div>
    <div>
      <button
        className="snapguidist__button"
        disabled={false}
        onClick={[Function]}
      >
        Update
      </button>
      <button
        className="snapguidist__button"
        disabled={false}
        onClick={[Function]}
      >
        <span
          className="snapguidist__arrow"
        >
          ▼
        </span>
      </button>
    </div>
    <div>
      <div>
        <Code
          label="Actual"
          value="actual"
        />
        <Code
          label="Expected"
          value="expected"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`works when fail, with diff 1`] = `
<div
  className="snapguidist__test snapguidist__test--fail"
>
  <div>
    <div>
      <button
        className="snapguidist__button"
        disabled={false}
        onClick={[Function]}
      >
        Update
      </button>
      <button
        className="snapguidist__button"
        disabled={false}
        onClick={[Function]}
      >
        <span
          className="snapguidist__arrow"
        >
          ▼
        </span>
      </button>
    </div>
    <div>
      <div>
        <Code
          label="Actual"
          value="actual"
        />
        <Code
          label="Expected"
          value="expected"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`works when fail, without diff) 1`] = `
<div
  className="snapguidist__test snapguidist__test--fail"
>
  <div>
    <div>
      <button
        className="snapguidist__button"
        disabled={false}
        onClick={[Function]}
      >
        Update
      </button>
      <button
        className="snapguidist__button"
        disabled={false}
        onClick={[Function]}
      >
        <span
          className="snapguidist__arrow"
        >
          ▼
        </span>
      </button>
    </div>
    <div>
      <Code
        diff={true}
        label="Difference"
        value="diff"
      />
    </div>
  </div>
</div>
`;

exports[`works when is fetching 1`] = `
<div
  className="snapguidist__test snapguidist__test--fail"
>
  <div>
    <div>
      <button
        className="snapguidist__button"
        disabled={true}
        onClick={[Function]}
      >
        Update
      </button>
      <button
        className="snapguidist__button"
        disabled={true}
        onClick={[Function]}
      >
        <span
          className="snapguidist__arrow"
        >
          ▼
        </span>
      </button>
    </div>
    <div>
      <div>
        <Code
          label="Actual"
          value="actual"
        />
        <Code
          label="Expected"
          value="expected"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`works when pass 1`] = `
<div
  className="snapguidist__test snapguidist__test--pass"
/>
`;
